pandas/tests/series/test_analytics.py

```diff
@@ -1455,6 +1455,16 @@ def test_nsmallest_nlargest(self):
         expected = s.sort_values().head(3)
         assert_series_equal(result, expected)
 
+        # GH 15297
+        s = Series([1] * 5, index=[1, 2, 3, 4, 5])
+        expected = Series([1] * 3, index=[1, 2, 3])
```

Comment:
add to the doc-string for both Series/DataFrame what the policy for dealing with duplicates is. IOW, we are going to always return n (or less values), so the duplicated portion would be in order of the index (see if you can reword!)